Home_Work_20/DeckSelectionHelper.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+#include <vector>
+#include <algorithm>
+#include <random>
+#include <iostream>
+
+
+
+template <class T>
+class DeckSelectionHelper
+{
+public:
+
+	T* operator()(std::vector<T*>& entityList, std::vector<T*>& entityExcludeList)
+	{
+		if (entityList.size()== entityExcludeList.size())
+		{
+			entityExcludeList.clear();
+		}
+
+		T* generatedEntity = nullptr;
+
+		while(generatedEntity == nullptr) 
+		{
+			
+			std::shuffle(entityList.begin(), entityList.end(), std::random_device());
```

Comment:
It looks like the shuffle will be quite often executed in the loop closer to the end of the deck.
What if instead of `excludeList` use `availableList` ? 
Then you will init once with shuffle, and just unconditionally get the front element.
When this list gets empty - re-init it once again. No more shuffle every now and then.

Reply:
Also, this list can be part of this helper itself. No need to hold a copy of it in all decks as they are not used there except for this helper